common/src/main/java/com/wynntils/core/crowdsource/CrowdSourcedData.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright © Wynntils 2023.
+ * This file is released under LGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.core.crowdsource;
+
+import com.google.gson.JsonArray;
+import com.google.gson.JsonDeserializationContext;
+import com.google.gson.JsonDeserializer;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParseException;
+import com.google.gson.JsonPrimitive;
+import com.wynntils.core.crowdsource.type.CrowdSourcedDataGameVersion;
+import com.wynntils.core.crowdsource.type.CrowdSourcedDataType;
+import java.lang.reflect.Type;
+import java.util.Map;
+import java.util.Set;
+import java.util.TreeMap;
+import java.util.TreeSet;
+import java.util.stream.Collectors;
+
+/**
+ * A holder class for crowd sourced data.
+ * This class is only meant to be modified by the {@link CrowdSourcedDataManager}.
+ * If you need to access crowd sourced data, use the {@link CrowdSourcedDataManager}.
+ */
+public class CrowdSourcedData {
+    private final Map<CrowdSourcedDataGameVersion, Map<CrowdSourcedDataType, Set<Object>>> data = new TreeMap<>();
+
+    CrowdSourcedData() {}
+
+    CrowdSourcedData(Map<CrowdSourcedDataGameVersion, Map<CrowdSourcedDataType, Set<Object>>> deserializedData) {
```

Comment:
I also don't see why you would need to code in the version as an argument here. We should not really persist crowd-source data between mod restarts, but send it off to Athena, right? 

(Which, in fact, I can't really see that you are doing -- is that planned as a future "enhancement"?)